### ISSUE TITLE
Bug in sprite_list, solitaire example broken

### ISSUE THREAD

**Issue by lwanger** [pythonarcade/arcade#824]
## Bug Report

There are two bugs in the sprite_list::__setitem__ method that I found while working through the solitaire example (https://arcade.academy/tutorials/card_game/index.html). Solitaire has a couple of places where a line like the following occurs (pull_to_top and shuffling):

```py
window.card_list[1], window.card_list[2] = window.card_list[2], window.card_list[1]
````

For simplicity I can reproduce the bug with the line:
```py
window.card_list[1] = window.card_list[2]
```

### Actual behavior:

First, __setitem__ causes a ValueError exception on the line:
```py
value.sprite_lists.remove(item_to_be_removed)
```

This should be:
```py
    self.sprite_lists.remove(item_to_be_removed)
```

Also, the following line shortens the length of the list by one (sets window.card_list[1] to the value in index two but does not add the value from index 1 back in -- shortening the length of the list by 1):

```py
window.card_list[1] = window.card_list[2]
```

This eventually causes the solitaire example to stop with an IndexError exception.

### Expected behavior:

Should swap the two values in the list, keeping the list the same length.

### Steps to reproduce/example code:

```py
import arcade

class Card(arcade.Sprite):
    def __init__(self, value):
        self.value = value
        super().__init__(None, 1.0)

    def __repr__(self):
        return f"Card={self.value}"

class MyGame(arcade.Window):
    def __init__(self):
        super().__init__(1000, 768, "shuffle test")
        self.card_list = None

    def setup(self):
        self.card_list = arcade.SpriteList()
        for i in range(5):
            self.card_list.append(Card(i))

if __name__ == "__main__":
    window = MyGame()
    window.setup()

    print(f"length of card_list = {len(window.card_list)},  card_list={window.card_list.sprite_list}")
    window.card_list[1] = window.card_list[2]
    print(f"length of card_list = {len(window.card_list)}, card_list={window.card_list.sprite_list}")
```

### What would it help with?

Gettting the tutorial code to work

### Where is it located?

sprite_list.py::__setitem__

**Comment by shernd0n** [comment]
I tried the completed code from step 11 and also got failures on the shuffle. But I'm getting:
`ValueError: list.remove(x): x not in list`
I then started over with a new file and followed along for every step. Similarly, I'm seeing errors with value.sprite_lists.remove from the pull_to_top function now that I've completed as far as step 3 (implement drag & drop). 

However, it does not seem to be impacting the on-screen function, but my terminal becomes full of these errors.

I do hit that hard stop with the shuffle function, but I made my deck shuffled from the beginning instead in the setup:

# Create every card

```py
for card_suit in CARD_SUITS:
    for card_value in CARD_VALUES:
        card = Card(card_suit, card_value, CARD_SCALE)
        card.position = START_X, BOTTOM_Y
        self.card_list.append(card)
```
to:

```py
# Create every card
for card_suit in CARD_SUITS:
    for card_value in CARD_VALUES:
        card = Card(card_suit, card_value, CARD_SCALE)
        card.position = START_X, BOTTOM_Y
        self.card_list.insert(random.randint(0,len(self.card_list)),card)
```

and then I commented out the three lines of code for the shuffle.

**Comment by lwanger** [comment]
I added a new shuffle method to sprite_list using the shuffle method in the standard library random module. The code becomes very simple. Replace all of those lines with:
```py
self.card_list.shuffle()
```
Does that work for you?

**Comment by shernd0n** [comment]
I tried that when I first got the ValueError, but it still wouldn't run the complete step 11 code. I believe I probably mistook the error because of the other instances of the pull_to_top error. It seems I went on a side-quest to fix the shuffle once I was seeing the pull_to_top not having any impact at step 3. 

I just confirmed, with `random.shuffle(self.card_list)` but get the same ValueError message. So far I haven't found a way to shuffle the completed deck, only build the deck in a random way.

**Comment by lwanger** [comment]
random.shuffle(self.card_list) won't work. Instead, pull the latest development branch code and use:

self.card_list.shuffle()

The random.shuffle method does not work on SpriteList objects directly.

**Comment by shernd0n** [comment]
I did find a 2.6.0_dev1, but it's way beyond my skill level to implement that. Thanks for the explanation on the SpriteList objects. My daughter and I had a working Hearts game that was text-only that we tried to get working in pygame. We were looking that over again this weekend and I thought there must be a better way and finally came across the arcade module. I should probably start on something more basic, or try to convert her coin pick-up game that was mostly working.
